build-system/pr-check/local-tests.js

```diff
@@ -39,54 +39,52 @@ const timedExecOrDie =
 function main() {
   const startTime = startTimer(FILENAME, FILENAME);
   const buildTargets = determineBuildTargets();
-  downloadBuildOutput(FILENAME);
 
   if (!isTravisPullRequestBuild()) {
+    downloadBuildOutput(FILENAME);
     timedExecOrDie('gulp test --integration --nobuild --coverage');
     timedExecOrDie('gulp test --unit --nobuild --headless --coverage');
     timedExecOrDie('gulp test --dev_dashboard --nobuild');
     //TODO(estherkim): turn on when stabilized :)
     //timedExecOrDie('gulp e2e --nobuild');
   } else {
     printChangeSummary(FILENAME);
-    let ranTests = false;
 
-    if (buildTargets.has('RUNTIME') ||
-        buildTargets.has('BUILD_SYSTEM') ||
-        buildTargets.has('UNIT_TEST')) {
-
-      timedExecOrDie('gulp test --nobuild --headless --local-changes');
-      ranTests = true;
-    }
-
-    if (buildTargets.has('RUNTIME') ||
-        buildTargets.has('BUILD_SYSTEM') ||
-        buildTargets.has('INTEGRATION_TEST')) {
-
-      timedExecOrDie('gulp test --integration --nobuild --headless --coverage');
-      ranTests = true;
-    }
-
-    if (buildTargets.has('RUNTIME') ||
-        buildTargets.has('BUILD_SYSTEM') ||
-        buildTargets.has('UNIT_TEST')) {
-
-      timedExecOrDie('gulp test --unit --nobuild --headless --coverage');
-      //TODO(estherkim): turn on when stabilized :)
-      //timedExecOrDie('gulp e2e --nobuild');
-      ranTests = true;
-    }
-
-    if (buildTargets.has('DEV_DASHBOARD')) {
-      timedExecOrDie('gulp test --dev_dashboard --nobuild');
-      ranTests = true;
-    }
-
-    if (!ranTests) {
+    if (!(buildTargets.has('RUNTIME') ||
+          buildTargets.has('BUILD_SYSTEM') ||
+          buildTargets.has('UNIT_TEST') ||
+          buildTargets.has('INTEGRATION_TEST') ||
+          buildTargets.has('DEV_DASHBOARD'))) {
       console.log(
           `${FILELOGPREFIX} Skipping unit and integration tests because ` +
           'this commit not affect the runtime, build system, ' +
           'unit test files, integration test files, or the dev dashboard.');
```

Comment:
Nit: Early return here would mean you could unnest the else clause below.

Reply:
Done - thanks for the quick review.